core/src/integrationTest/java/tech/pegasys/web3signer/core/jsonrpcproxy/IntegrationTestBase.java

```diff
@@ -0,0 +1,386 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.Resources;
+import io.restassured.RestAssured;
+import io.restassured.config.HttpClientConfig;
+import io.restassured.response.Response;
+import io.restassured.specification.RequestSpecification;
+import io.vertx.core.Vertx;
+import org.apache.http.params.CoreConnectionPNames;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockserver.integration.ClientAndServer;
+import org.mockserver.model.JsonBody;
+import org.mockserver.model.RegexBody;
+import org.web3j.crypto.Credentials;
+import org.web3j.crypto.WalletUtils;
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.JsonRpc2_0Web3j;
+import org.web3j.protocol.eea.Eea;
+import org.web3j.protocol.eea.JsonRpc2_0Eea;
+
+public class IntegrationTestBase {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private static final String PORTS_FILENAME = "web3signer.ports";
+  private static final String HTTP_PORT_KEY = "http-port";
+  private static final String LOCALHOST = "127.0.0.1";
+  public static final long DEFAULT_CHAIN_ID = 9;
+  public static final int DEFAULT_ID = 77;
+
+  static final String MALFORMED_JSON = "{Bad Json: {{{}";
+
+  private static Vertx vertx;
+  private static Eth1Runner runner;
+  static ClientAndServer clientAndServer;
+  static Credentials credentials;
+
+  private JsonRpc2_0Web3j jsonRpc;
+  private JsonRpc2_0Eea eeaJsonRpc;
+
+  protected final EthRequestFactory request = new EthRequestFactory();
+  protected final EthResponseFactory response = new EthResponseFactory();
+
+  static String unlockedAccount;
+
+  private static final Duration downstreamTimeout = Duration.ofSeconds(1);
+
+  @TempDir static Path dataPath;
+  @TempDir static Path keyConfigPath;
+
+  static void setupWeb3Signer(final long chainId) throws Exception {
+    setupWeb3Signer(chainId, "");
+  }
+
+  static void setupWeb3Signer(final long chainId, final String downstreamHttpRequestPath)
+      throws Exception {
+    setupWeb3Signer(chainId, downstreamHttpRequestPath, List.of("sample.com"));
+  }
+
+  static void setupWeb3Signer(
+      final long chainId,
+      final String downstreamHttpRequestPath,
+      final List<String> allowedCorsOrigin)
+      throws Exception {
+    clientAndServer = startClientAndServer();
+
+    final File keyFile = createKeyFile();
+    final File passwordFile = createFile("password");
+    credentials = WalletUtils.loadCredentials("password", keyFile);
+
+    final Eth1AddressSignerProvider transactionSignerProvider =
+        new Eth1AddressSignerProvider(new SingleSignerProvider(signer(keyFile, passwordFile)));
+
+    // Force TransactionDeserialisation to fail
+    final ObjectMapper jsonObjectMapper = new ObjectMapper();
```

Comment:
why do we need to force this to fail?

Reply:
In fact, does the jsonObjectMapper even get used anywhere?

Reply:
not needed anymore. removed it